pkg/operator/nfs/provisioner.go

```diff
@@ -86,67 +113,78 @@ func (p *nfsProvisioner) Provision(options controller.ProvisionOptions) (*v1.Per
 				v1.ResourceName(v1.ResourceStorage): options.PVC.Spec.Resources.Requests[v1.ResourceName(v1.ResourceStorage)],
 			},
 			PersistentVolumeSource: v1.PersistentVolumeSource{
-				NFS: nfsVolumeSource,
+				NFS: &v1.NFSVolumeSource{
+					Server:   nfsserversvc.Spec.ClusterIP,
+					Path:     "/" + fullPath,
+					ReadOnly: false,
+				},
 			},
 		},
 	}
+
 	return pv, nil
 }
 
-func (p *nfsProvisioner) Delete(volume *v1.PersistentVolume) error {
-	return nil
+func (p *Provisioner) Delete(volume *v1.PersistentVolume) error {
+	path := volume.Spec.PersistentVolumeSource.NFS.Path
+	pvName := filepath.Base(path)
+
+	sc, err := p.storageClassForPV(volume)
+	if err != nil {
+		return err
+	}
+
+	serverName, present := sc.Parameters["nfsServerName"]
+	if !present {
+		return errors.Errorf("NFS share Path not found in the storageclass: %v", sc.GetName())
+	}
+
+	serverNamespace, present := sc.Parameters["nfsServerNamespace"]
+	if !present {
+		return errors.Errorf("NFS share Path not found in the storageclass: %v", sc.GetName())
+	}
+
+	exportName, present := sc.Parameters["exportName"]
+	if !present {
+		return errors.Errorf("NFS share Path not found in the storageclass: %v", sc.GetName())
+	}
+
+	nfsserver, err := p.rookClient.NfsV1alpha1().NFSServers(serverNamespace).Get(serverName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	var exportPath string
+	for _, export := range nfsserver.Spec.Exports {
+		if export.Name == exportName {
+			exportPath = filepath.Join(mountPath, export.PersistentVolumeClaim.ClaimName)
```

Comment:
how about logging something if exportName is not matched, that would be helpful for debugging